include/exadg/functions_and_boundary_conditions/interface_coupling.h

```diff
@@ -533,175 +140,69 @@ class InterfaceCoupling
+                                std::vector<Tensor<rank, dim, double>>(1));
 
       /*
        * 2. Communication: receive and cache quadrature points of other ranks,
        *    redundantly store own q-points (those that are needed)
        */
-      map_communicator.emplace(quadrature,
-                               InterfaceCommunicator<dim, dim>(array_q_points_dst,
-                                                               *triangulation_src,
-                                                               *mapping_src,
-                                                               tolerance,
-                                                               marked_vertices,
-                                                               cache_src));
-
-      InterfaceCommunicator<dim, dim> & communicator = map_communicator.find(quadrature)->second;
-
-      mpi_map_q_points_src = communicator.get_remote_quadrature_points();
-
-      communicator.init_solution_values(array_solution_dst);
-
-      communicator.init_remote_solution_values(mpi_map_solution_src);
-
-
-      /*
-       * 3. Compute dof indices and shape values for all quadrature points
-       */
-      typename Triangulation<dim, dim>::active_cell_iterator cell_hint =
-        triangulation_src->begin_active();
-
-      for(auto it : mpi_map_q_points_src)
-      {
-        mpi_rank const          proc               = it.first;
-        ArrayQuadraturePoints & array_q_points_src = it.second;
-
-        mpi_map_cache_src.emplace(proc, ArrayVectorCache());
-        ArrayVectorCache & array_cache_src = mpi_map_cache_src.find(proc)->second;
-        array_cache_src.resize(array_q_points_src.size());
-
-        for(types::global_dof_index q = 0; q < array_q_points_src.size(); ++q)
-        {
-          auto adjacent_cells =
-            find_all_active_cells_around_point(*mapping_src,
-                                               dof_handler_src->get_triangulation(),
-                                               array_q_points_src[q],
-                                               tolerance,
-                                               cell_hint,
-                                               marked_vertices,
-                                               *cache_src);
-
-          array_cache_src[q] = get_dof_indices_and_shape_values(adjacent_cells,
-                                                                *dof_handler_src,
-                                                                *mapping_src,
-                                                                *dof_vector_src_double_ptr);
-
-          AssertThrow(array_cache_src[q].size() > 0,
-                      ExcMessage("No adjacent points have been found."));
-        }
-      }
-
-
-      /*
-       * 4. Communication: transfer results back to dst-side
-       */
-      communicator.process(mpi_map_solution_src, array_solution_dst, 11 + quadrature);
```

Comment:
I am not sure why this communication step is performed here.

Reply:
I am also not sure and don't remember where this comes from. For example, I wonder about `11`? @peterrum you probably know best what `communicator.process()` was doing.